src/mgxs_data.F90

```diff
@@ -0,0 +1,705 @@
+            representation = MGXS_ISOTROPIC
+          end if
+
+          ! Now allocate accordingly
+          select case(representation)
+          case(MGXS_ISOTROPIC)
+            allocate(NuclideIso :: nuclides_MG(i_nuclide) % obj)
+          case(MGXS_ANGLE)
+            allocate(NuclideAngle :: nuclides_MG(i_nuclide) % obj)
+          end select
+
+          ! Now read in the data specific to the type we just declared
+          call NuclideMG_init(nuclides_MG(i_nuclide) % obj, node_xsdata, &
+                               energy_groups, get_kfiss, get_fiss, error_code, &
+                               error_text)
+
+          ! Keep track of what listing is associated with this nuclide
+          nuclides_MG(i_nuclide) % obj % listing = i_listing
+
+          ! Handle any errors
+          if (error_code /= 0) then
+            call fatal_error(trim(error_text))
+          end if
+
+          ! Add name and alias to dictionary
+          call already_read % add(name)
+          call already_read % add(alias)
+        end if
+      end do NUCLIDE_LOOP
+    end do MATERIAL_LOOP
+
+    ! Avoid some valgrind leak errors
+    call already_read % clear()
+
+    ! Loop around material
+    MATERIAL_LOOP3: do i = 1, n_materials
+
+      ! Get material
+      mat => materials(i)
+
+      ! Loop around nuclides in material
+      NUCLIDE_LOOP2: do j = 1, mat % n_nuclides
+        ! Is this fissionable?
+        if (nuclides_MG(mat % nuclide(j)) % obj % fissionable) then
+          mat % fissionable = .true.
+        end if
+        if (mat % fissionable) then
+          exit NUCLIDE_LOOP2
+        end if
+
+      end do NUCLIDE_LOOP2
+    end do MATERIAL_LOOP3
+
+  end subroutine read_mgxs
+
+!===============================================================================
+! SAME_NUCLIDE_LIST creates a linked list for each nuclide containing the
+! indices in the nuclides array of all other instances of that nuclide.  For
+! example, the same nuclide may exist at multiple temperatures resulting
+! in multiple entries in the nuclides array for a single zaid number.
+!===============================================================================
+
+  subroutine same_NuclideMG_list()
+
+    integer :: i ! index in nuclides array
+    integer :: j ! index in nuclides array
+
+    do i = 1, n_nuclides_total
+      do j = 1, n_nuclides_total
+        if (nuclides_MG(i) % obj % zaid == nuclides_MG(j) % obj % zaid) then
+          call nuclides_MG(i) % obj % nuc_list % push_back(j)
+        end if
+      end do
+    end do
+
+  end subroutine same_NuclideMG_list
+
+!===============================================================================
+! NUCLIDE_*_INIT reads in the data from the XML file, as already accessed
+!===============================================================================
+
+    subroutine NuclideMG_init(this, node_xsdata, groups, get_kfiss, get_fiss, &
+                                  error_code, error_text)
```

Comment:
I'm stretching far in to the annals here, but at the time when I wrote this, these routines needed to pull in xml_interface, which pulled in error, which pulled in global.  That's not the case anymore (since your secondary distribution refactor?), so it should be fine to move these back to a proper home.  That also means I dont need to have the error throwing system I had going on as well and can instead just call fatal_error (your next comment)